### PR TITLE
feat(container.signature): add ContainerSignatureValidationService APIs

### PR DESCRIPTION
**Pull request by mattdibi**
The purpose of this PR is to define the APIs for future implementations of a `ContainerSignatureValidationService`, whose goal will be (you guessed it) to verify container image signatures.

Our plan is to provide an API generic enough that all major tools will be compatible:
- [Sigstore's Cosign](https://docs.sigstore.dev/signing/quickstart/)
- [Docker Content Trust (a.k.a. Notary V1)](https://docs.docker.com/engine/security/trust/)
- [Notation (a.k.a. Notary V2)](https://notaryproject.dev/)

The `ContainerSignatureValidationService` implementations will be used by the `ContainerOrchestrationService` to validate the authenticity of the requested container images to further improve the overall security of the platform Kura is running on.

### CODE REVIEW

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+import java.util.Optional;
+import org.eclipse.kura.configuration.Password;
+import org.eclipse.kura.container.orchestration.ImageInstanceDescriptor;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Interface representing a service for validating the signature of a container image
+ *
+ * @since 2.3
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java]: I'm not sure the version is correct. It should match the bundle version as far as I know

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ * @noimplement This interface is not intended to be implemented by clients.
+ */
+@ProviderType
+public interface ContainerSignatureValidationService {
+
+    /**
+     * Verifies the signature of a container image using the provided trust anchor. The trust anchor format depends on
+     * the signature format. For example, if the signature was generated with Cosing, the trust anchor is a ECDSA public
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java]:
```suggestion
     * the signature format. For example, if the signature was generated with Cosign, the trust anchor is a ECDSA public
```

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+     * key in PEM format. Other signature formats may require different trust anchors.
+     *
+     * If the signature is not included in a transparency log, the verification will fail unless the
+     * verifyInTransparencyLog is set to false.
+     *
+     * If the image is signed with a different protocol the verification will fail.
+     *
+     * If the device running the verification has no internet access, the verification will fail.
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java]: Can we state this or it is implementation specific?

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+     *            transparency log. Artifacts cannot be publicly verified when not included in a log.
+     * @param registryUsername
+     *            Optional username for registry authentication. If the registry requires authentication,
+     *            both username and password must be provided.
+     * @param registryPassword
+     *            Optional password for registry authentication. If the registry requires authentication,
+     *            both username and password must be provided.
+     * @return
+     */
+    public boolean verify(String imageName, String imageReference, String trustAnchor, boolean verifyInTransparencyLog,
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java:95]: better to overload to not have the Optional in the parameters?

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java (hunk 3 above)
Comment (mattdibi) [reply kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java]: This is our design choice: if we don't have internet access we cannot perform the verification and thus return `false`, regardless of the implementation.

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java (hunk 1 above)
Comment (mattdibi) [reply kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java]: Should be addressed in 43d61a230ecfa9871a9bc18ab9c88653fc9c7363

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java (hunk 4 above)
Comment (mattdibi) [reply kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java:95]: Addressed in 8252906ec250f8b58cd780cf1a8170e5fb744239

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java (hunk 3 above)
Comment (mattdibi) [reply kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java]: Addressed in 4ff066f1ed4b4ab2b2764608ed33b3cf95f972c7 as per our discussion

**Review by MMaiero — APPROVED** [review]

**Pull request merged**